extensions/common/context.cc

```diff
@@ -181,16 +184,16 @@ void populateHTTPRequestInfo(bool outbound, RequestInfo* request_info) {
   std::string cluster_name = "";
   getStringValue({"cluster_name"}, &cluster_name);
   extractFqdn(cluster_name, &request_info->destination_service_host);
-  if (request_info->destination_service_host.empty()) {
-    // fallback to host header.
+  if (!request_info->destination_service_host.empty()) {
+    // cluster name follows Istio convention, so extract out service name.
+    extractServiceName(request_info->destination_service_host,
+                       &request_info->destination_service_name);
+  } else if (use_host_header_fallback) {
```

Comment:
Can we make this if (user_host_header_fallback && !cluster_name.empty())?

Reply:
Actually how about just `if (!cluster_name.empty())` ?

Reply:
what is the else clause ? Do you want to just set it to cluster name then ?
For non-pilot control planes it will not be meaningful.

Reply:
No, the else clause still falls back to request header, but only if there is a known route. My understanding is that if there is a known route, request host must be some constrained value, instead of random, no?

Reply:
known route or unknown route is orthogonal to using host header.

You can have a known route with a wildcard match, and you still need to decide if it ok to use the host header.

Reply:
I see. Thanks!